lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -101,24 +94,15 @@ def add_top_level_redirection_headers(url: nil, ignore_response_code: false)
           end
         end
 
-        url ||= login_url_with_optional_shop
+        url = login_url_with_optional_shop
 
         ShopifyApp::Logger.debug("Setting Reauthorize-Url to #{url}")
-        response.set_header("X-Shopify-API-Request-Failure-Reauthorize", "1")
-        response.set_header("X-Shopify-API-Request-Failure-Reauthorize-Url", url)
+        RedirectForEmbedded.add_app_bridge_redirect_url_header(url, response)
       end
     end
 
     protected
 
-    def jwt_shopify_domain
-      request.env["jwt.shopify_domain"]
-    end
-
-    def jwt_shopify_user_id
-      request.env["jwt.shopify_user_id"]
-    end
```

Comment:
Extracted these 'jwt' helpers to be inside `WithShopifyIdToken` concern for reusability